plugins/builds/index.js

```diff
@@ -718,11 +719,15 @@ const buildsPlugin = {
                         buildConfig.jobId = nextJob.id;
                         if (!isExternal) {
                             buildConfig.eventId = event.id;
-                        } else {
+                        } else if (hoek.reach(pipelineJoinData[pid], 'event')) {
```

Comment:
Can you assign directly?
`buildConfig.eventId = hoek.reach(pipelineJoinData[pid], 'event.id')`

https://github.com/hapijs/hoek/blob/master/API.md#reachobj-chain-options